### PR TITLE
Support "Uncategorized" extensions

### LINKED ISSUE #21350: Support "Uncategorized" extensions
### Description

Currently extensions without category are not processed and not visible in code.quarkus.io

### Implementation ideas

Use "uncategorized" as category here (if empty): https://github.com/quarkusio/quarkus/blob/main/independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/catalog/processor/CatalogProcessor.java#L25

Add "Uncategorized" as category last in `orderedCategories` here: https://github.com/quarkusio/quarkus/blob/main/independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/catalog/processor/CatalogProcessor.java#L33

### PR DESCRIPTION
**Pull request by geekiyer**
- Closes #21350

### CODE REVIEW

**Comment by ia3andy** [comment]
@geekiyer FYI: https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history

(and use `git push --force-with-lease` to make sure you don't erase remote commits)

**Review by ia3andy — APPROVED** [review]
LGTM

**Comment by ia3andy** [comment]
Thanks @geekiyer 👍

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/catalog/processor/CatalogProcessor.java [hunk 1]
```diff
@@ -22,6 +26,10 @@ public static CatalogProcessor of(ExtensionCatalog catalog) {
         final Map<String, List<Extension>> extsByCategory = new HashMap<>(catalog.getCategories().size());
         for (Extension e : catalog.getExtensions()) {
             List<String> categories = ExtensionProcessor.of(e).getCategories();
+            if (categories.isEmpty()) {
+                extsByCategory.put(UNCATEGORIZED_ID, new ArrayList<>());
```
Comment (gastaldi) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/catalog/processor/CatalogProcessor.java:30]: This seems to discard existing uncategorized extensions, is that intended?

Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/catalog/processor/CatalogProcessor.java:30]: @gastaldi why would it discard them?

Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/catalog/processor/CatalogProcessor.java:30]: it's the opposite, it's making them visible

Comment (gastaldi) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/catalog/processor/CatalogProcessor.java:30]: Well, it's in a loop, It seems like the `extsByCategory` map would already have an entry in the second iteration, but I might have been missing something

Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/catalog/processor/CatalogProcessor.java:30]: Ah snap, I missed that, I thought it was adding to the categories list 😬. I will push a patch now

Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/platform/catalog/processor/CatalogProcessor.java:30]: https://github.com/quarkusio/quarkus/pull/21522

**Pull request merged**